javascript/resources/tools/autobuild.sh

```diff
@@ -4,12 +4,33 @@ set -eu
 
 jvm_args=-Xss16m
 
-# If CODEQL_RAM is set, use half for Java and half for TS.
+# If CODEQL_RAM is set, try to automatically calculate how much memory is available for TS and Java
+# if no explicit values are set for them.
 if [ -n "${CODEQL_RAM:-}" ] ; then
     half_ram="$(( CODEQL_RAM / 2 ))"
-    LGTM_TYPESCRIPT_RAM="$half_ram"
+
+    # If either CODEQL_JAVASCRIPT_TS_MEMORY or CODEQL_JAVASCRIPT_JVM_MAX_MEMORY is set,
+    # calculate the other by subtracting from CODEQL_RAM. If neither is set, then use
+    # half of the available CODEQL_RAM for each.
+    if [ -n "${CODEQL_JAVASCRIPT_TS_MEMORY:-}" ] && [ -z "${CODEQL_JAVASCRIPT_JVM_MAX_MEMORY}" ] ; then
```

Comment:
Let's define these environment variables as extractor options by adding them to:
https://github.com/github/codeql/blob/988d0671bb164746777ccb0fd4f856a2c74f0d3a/javascript/resources/codeql-extractor.yml#L64-L94

Reply:
Would adding them there require us to rename them so that they start with `CODEQL_EXTRACTOR_JAVASCRIPT_OPTION`?

Reply:
Yes they would need to have that prefix,